DnsService/DnsService.cs

```diff
@@ -34,10 +34,25 @@ public DnsService()
 
         protected override void OnStart(string[] args)
         {
-            _service = new WebService(null, new Uri("https://go.technitium.com/?id=22"));
+            _service = new WebService(getConfigFolder(), new Uri("https://go.technitium.com/?id=22"));
             _service.Start();
         }
 
+        /// <summary>
+        /// If the service file path has the configuration files, use it, otherwise use the localapppath folder.
+        /// </summary>
+        /// <returns>The configuration path.</returns>
+        private string getConfigFolder()
+        {
+            string userConfigFolder = String.Concat(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), @"\Technitium\DNSServer");
+
+            if (System.IO.Directory.Exists(Environment.CurrentDirectory))
```

Comment:
Can you give an example of a situation when `Environment.CurrentDirectory` won't exist?

If `Environment.CurrentDirectory` is set to the process's executable directory when the process starts, then we know that directory must exist.

When changing the `Environment.CurrentDirectory`, such as `Environment.CurrentDirectory = "C:\\foobar\\";`, the specified directory must exist or it will throw an exception.

If you successfully set `Environment.CurrentDirectory` to some empty folder, for example, and then try to delete that folder, you won't be able to because a handle to it remains in use by the process.

So how else might it be possible for `Environment.CurrentDirectory` to not exist? If it always exists, then it means the code for returning the `userConfigFolder` is never reached.

Speaking of, the XML comment for this method is:
> If the service file path ***has the configuration files***, use it...

To me, when reading that comment, I expected to find code that looked for the existence of some configuration files, like these:
![image](https://user-images.githubusercontent.com/22459191/110694201-8ec82380-81a5-11eb-95ec-30eb283e3091.png)

But the code is only checking that a directory exists, not files.

Reply:
That code no longer exists. See the `develop` branch code.